inc/elasticsearch_packages/namespace.php

```diff
@@ -0,0 +1,432 @@
+<?php
+/**
+ * AES Package File Integration.
+ *
+ * @package altis/cloud
+ */
+
+namespace Altis\Cloud\Elasticsearch_Packages;
+
+use Altis;
+use Aws\ElasticsearchService\ElasticsearchServiceClient;
+use Exception;
+use WP_Error;
+
+/**
+ * Hook into search module package support.
+ *
+ * @return void
+ */
+function setup() {
+	// Check the current environment and bail if it isn't AWS.
+	if ( ! in_array( Altis\get_environment_architecture(), [ 'ec2', 'ecs' ], true ) ) {
+		return;
+	}
+
+	// Add a short time out cron hook.
+	// phpcs:ignore WordPress.WP.CronInterval.ChangeDetected
+	add_filter( 'cron_schedules', __NAMESPACE__ . '\\cron_schedules' );
+
+	// Hook into search module.
+	add_filter( 'altis.search.packages_dir', __NAMESPACE__ . '\\packages_dir', 9 );
+	add_filter( 'altis.search.create_package_id', __NAMESPACE__ . '\\create_package_id', 10, 4 );
+	add_filter( 'altis.search.get_package_id', __NAMESPACE__ . '\\get_package_id', 10, 2 );
+	add_action( 'altis.search.check_package_status', __NAMESPACE__ . '\\on_check_package_status', 10, 3 );
+	add_action( 'altis.search.delete_package', __NAMESPACE__ . '\\delete_package', 10, 2 );
+	add_action( 'altis.search.deleted_package', __NAMESPACE__ . '\\on_deleted_package', 10, 4 );
+	add_action( 'altis.search.updated_all_index_settings', __NAMESPACE__ . '\\on_update_index_settings' );
+
+	// Remove default packages updated hook so we don't try to update indexes until
+	// packages have been associated with the domain.
+	add_action( 'plugins_loaded', function () {
+		remove_action( 'altis.search.updated_packages', 'Altis\\Enhanced_Search\\Packages\\on_updated_packages', 10, 2 );
+	}, 20 );
+}
+
+/**
+ * Filter the cron schedules list.
+ *
+ * @param array $schedules The list of available cron schedules.
+ * @return array
+ */
+function cron_schedules( array $schedules ) : array {
+	$schedules['1minute'] = [
+		'display' => __( 'Every minute', 'altis' ),
+		'interval' => 60,
+	];
+	return $schedules;
+}
+
+/**
+ * Set the package upload directory to an S3 directory.
+ *
+ * @param string $path The directory to store packages in.
+ * @return string
+ */
+function packages_dir( string $path ) : string {
+	if ( ! defined( 'S3_UPLOADS_BUCKET' ) || empty( S3_UPLOADS_BUCKET ) ) {
+		return $path;
+	}
+
+	return sprintf( 's3://%s/es-packages', S3_UPLOADS_BUCKET );
+}
+
+/**
+ * Get Elasticsearch Service Client.
+ *
+ * @return ElasticsearchServiceClient
+ */
+function get_es_client() : ElasticsearchServiceClient {
+	return Altis\get_aws_sdk()->createElasticsearchService( [
+		'version' => '2015-01-01',
+	] );
+}
+
+/**
+ * Return the domain ID portion of the Elasticsearch Service host name.
+ */
+function get_elasticsearch_domain() : string {
+	return preg_replace( '#^search-([a-z][a-z0-9\-]+)-[a-z0-9]+\..*$#', '$1', ELASTICSEARCH_HOST );
```

Comment:
I think this is more idiomatic modern PHP, as it doesn't overlap with PHP's interpolated variable string syntax:

```suggestion
	return preg_replace( '#^search-([a-z][a-z0-9\-]+)-[a-z0-9]+\..*$#', '\\1', ELASTICSEARCH_HOST );
```

Reply:
From the php.net docs: https://www.php.net/manual/en/function.preg-replace.php

> replacement may contain references of the form \n or $n, with the latter form being the preferred one.

Reply:
Classic PHP. Welp, fine by me then!